libp2p/protocols/pubsub/gossipsub/behavior.nim

```diff
@@ -275,25 +275,19 @@ proc handleIWant*(g: GossipSub,
   var messages: seq[Message]
   if peer.score < g.parameters.gossipThreshold:
     trace "iwant: ignoring low score peer", peer, score = peer.score
-  elif peer.iWantBudget <= 0:
-    trace "iwant: ignoring out of budget peer", peer, score = peer.score
   else:
-    let deIwants = iwants.deduplicate()
```

Comment:
I've remove deduplication since this now happens naturally, though the computation is a tad more expensive